### PR TITLE
Add libsigcpp 2.10.x

### PR DESCRIPTION
**Pull request by gunmetal313**
Specify library name and version:  **libsigcpp/2.10.x**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

# TODO

- [ ] add SIGCXX_*_VERSION define version
- [ ] fix problem with SIGC_DLL and shared/not shared mode

Not finished yet

### CODE REVIEW

**Comment by Croydon** [comment]
> add more versions 2.10.x

isn't the latest patch version enough?

File: recipes/libsigcpp/2.10.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import os
+from conans import ConanFile, CMake, tools
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    url = "https://github.com/libsigcplusplus/libsigcplusplus"
+    homepage = "https://libsigcplusplus.github.io/libsigcplusplus/"
+    author = "Conan Community"
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (Croydon) [inline recipes/libsigcpp/2.10.x/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py (hunk 6 above)
Comment (gunmetal313) [reply recipes/libsigcpp/2.10.x/test_package/conanfile.py]: Why should I remove this? https://github.com/conan-io/conan-center-index/blob/b8683bc6a9edb20bfcec2182388245c2abecc7d6/recipes/libjpeg/all/conanfile.py#L1 
Here it is present

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py (hunk 5 above)
Comment (gunmetal313) [reply recipes/libsigcpp/2.10.x/test_package/conanfile.py]: Why should I remove this? https://github.com/conan-io/conan-center-index/blob/master/recipes/libjpeg/all/conanfile.py#L1-L2
Here it is present

File: recipes/libsigcpp/2.10.x/test_package/conanfile.py (hunk 7 above)
Comment (gunmetal313) [reply recipes/libsigcpp/2.10.x/test_package/conanfile.py]: vim env comment was removed

File: recipes/libsigcpp/2.10.x/conanfile.py (hunk 2 above)
Comment (gunmetal313) [reply recipes/libsigcpp/2.10.x/conanfile.py]: Why should I remove this? https://github.com/conan-io/conan-center-index/blob/master/recipes/libjpeg/all/conanfile.py#L1-L2
Here it is present

File: recipes/libsigcpp/2.10.x/conanfile.py (hunk 1 above)
Comment (gunmetal313) [reply recipes/libsigcpp/2.10.x/conanfile.py]: Why should I remove this? https://github.com/conan-io/conan-center-index/blob/master/recipes/libjpeg/all/conanfile.py#L1-L2
Here it is present

Comment (Croydon) [reply recipes/libsigcpp/2.10.x/conanfile.py]: This is a policy which got adopted after libjpeg got merged. The next time somebody modifies libjpeg these lines need to be removed there as well.

Please install the latest version of the conan-center hook, so you get the warnings locally as well. Reference: https://github.com/conan-io/hooks/pull/94

CCI is currently under maintenance, but the Conan bot will complain about this as well (the Conan bot uses these hooks as well).

**Comment by gunmetal313** [comment]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl sigc::internal::slot_rep::slot_rep(void * (__cdecl*)(void *),void * (__cdecl*)(void *),void * (__cdecl*)(void *))" (__imp_??0slot_rep@internal@sigc@@QEAA@P6APEAXPEAX@Z11@Z) referenced in function "public: static void * __cdecl sigc::internal::typed_slot_rep<class sigc::pointer_functor1<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,void> >::dup(void *)" (?dup@?$typed_slot_rep@V?$pointer_functor1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@X@sigc@@@internal@sigc@@SAPEAXPEAX@Z) [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl sigc::internal::slot_do_bind::slot_do_bind(struct sigc::internal::slot_rep *)" (__imp_??0slot_do_bind@internal@sigc@@QEAA@PEAUslot_rep@12@@Z) referenced in function "public: static void * __cdecl sigc::internal::typed_slot_rep<class sigc::pointer_functor1<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,void> >::dup(void *)" (?dup@?$typed_slot_rep@V?$pointer_functor1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@X@sigc@@@internal@sigc@@SAPEAXPEAX@Z) [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl sigc::internal::slot_do_unbind::slot_do_unbind(struct sigc::internal::slot_rep *)" (__imp_??0slot_do_unbind@internal@sigc@@QEAA@PEAUslot_rep@12@@Z) referenced in function "public: static void * __cdecl sigc::internal::typed_slot_rep<class sigc::pointer_functor1<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,void> >::destroy(void *)" (?destroy@?$typed_slot_rep@V?$pointer_functor1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@X@sigc@@@internal@sigc@@SAPEAXPEAX@Z) [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl sigc::slot_base::empty(void)const " (__imp_?empty@slot_base@sigc@@QEBA_NXZ) referenced in function main [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl sigc::slot_base::blocked(void)const " (__imp_?blocked@slot_base@sigc@@QEBA_NXZ) referenced in function main [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]
test_package.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl sigc::internal::signal_exec::signal_exec(struct sigc::internal::signal_impl const *)" (__imp_??0signal_exec@internal@sigc@@QEAA@PEBUsignal_impl@12@@Z) referenced in function main [C:\projects\conan-center-index\recipes\libsigcpp\2.10.x\test_package\build\882726301a296ee1f7f1d98a3db2003150ea70a8\test_package.vcxproj]

errors on windows on shared mode now

File: recipes/libsigcpp/config.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+versions:
+  2.10.x:
```
Comment (SSE4) [inline recipes/libsigcpp/config.yml]:
```suggestion
  2.10.2:
```

**Comment by Croydon** [comment]
By the way, we do have a recipe for this library at Bincrafters as well: https://github.com/bincrafters/conan-sigcpp

**Pull request closed without merging**